cdap-docs/developers-manual/source/building-blocks/transaction-system.rst

```diff
@@ -335,9 +254,94 @@ the timeout in seconds to the ``execute()`` method::
 
 
+.. _transaction-system-conflict-detection:
+
+Levels of Conflict Detection
+============================
+Transactions providing ACID (atomicity, consistency, isolation, and durability) guarantees
+are useful in several applications where data accuracy is critical—examples include billing
+applications and computing click-through rates.
+
+However, transaction are not for free: the transaction system must track all the writes
+made by all transactions, and it must check transactions for conflicts before committing them.
+If conflicts are frequent, they will impact performance because the failed transactions
+have to be rolled back and reattempted.
+
+In some scenarios, you may want to fine-tune the manner in which a dataset participates in
+transactions:
+
+- Some applications—such as trending—might not need transactions for all writes, because
+  small inaccuracies have little effect on trends with great momentum. Applications that
+  do not strictly require accuracy can trade it for increased throughput by disabling
+  transactions for some datasets.
+- Some applications perform concurrent updates to the same row of a table, but typically
+  those updates do not strictly conflict with each other because they are on different
+  columns of the row. In this case it can make sense to increase the precision of conflict
+  detection by tracking changes at the column level instead of the row level.
+
+Both of these can be achieved by specifying a conflict detection level when the table is
+created. For example, in your application's ``configure()`` method::
+
+    Tables.createTable(getConfigurer(), "myTable", ConflictDetection.COLUMN);
+
+You have these options:
+
+- ``ConflictDetection.NONE`` to disable transactions for the table. None of the writes
+  performed on this table will participate in conflict detection. However, all writes
+  will still be rolled back in case of a transaction failure (the transaction may fail
+  for other reasons than a conflict on this table).
+- ``ConflictDetection.ROW`` to track writes at the row level. This means that two
+  concurrent transactions will cause a conflict if they write to the same row of the table,
+  even if the writes are for different columns. This is the default.
+- ``ConflictDetection.COLUMN`` to increase the precision of the conflict detection to
+  the column level: Two concurrent transactions can write to the same row without conflict,
+  as long as they write to disjoint sets of columns. This will increase the overhead for
+  each transaction, because the transaction system must track writes with greater detail.
+  But it can also greatly reduce the number of transaction conflicts, leading to improved
+  overall application throughput.
+
+  See the :ref:`UserProfile example <examples-user-profiles>`
+  for a sample use case.
+
+
+.. _transaction-system-transactions-mapreduce:
+
+Transactions and MapReduce
```

Comment:
Can we put this after "Using Transactions in Programs"? I almost feel like this should even be moved into the MapReduce chapter. Between "MapReduce and Datasets" and "MapReduce and Resources". And add a link to it here, same as we did for Spark.